git-tools/hooks/commit-msg

```diff
@@ -147,6 +147,15 @@ then
 	quit $ERR_LENGTH;
 fi
 
+# Check for CR/LF line breaks
+if grep -q $'\r$' "$1"
```

Comment:
Right now this would only detect \r at the end of a line, which is the classic windows line ending. CRLF should however be \r\n which means there probably needs to be an optional addition for \n at the end of the line between the \r and the $.

Reply:
That's what one would think, but consider this:

    $ echo test > unix
    $ echo test > dos
    $ unix2dos dos
    unix2dos: converting file dos to DOS format ...
    $ od -a unix
    0000000   t   e   s   t  nl
    0000005
    $ od -a dos
    0000000   t   e   s   t  cr  nl
    0000006
    $ if grep -q $'\r$' unix; then echo crlf; fi
    $ if grep -q $'\r$' dos; then echo crlf; fi
    crlf
    $ if grep -q $'\r\n$' unix; then echo crlf; fi
    crlf
    $ if grep -q $'\r\n$' dos; then echo crlf; fi
    crlf

Based on this, adding \n would be harmful.

Reply:
that's because grep works line by line and never consider \n :)